azurerm/resource_arm_analysis_services_server_test.go

```diff
@@ -0,0 +1,379 @@
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.0.range_end", "92.123.234.12"),
+				),
+			},
+			{
+				Config: config3,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAnalysisServicesServerExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "enable_power_bi_service", "true"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.#", "2"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.0.name", "test1"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.0.range_start", "92.123.234.11"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.0.range_end", "92.123.234.13"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.1.name", "test2"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.1.range_start", "226.202.187.57"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.1.range_end", "226.208.192.47"),
+				),
+			},
+		},
+	})
+}
+
+// Currently unsure how to test it as the email addresses need to exist in the AD
+//func TestAccAzureRMAnalysisServicesServer_adminUsers(t *testing.T) {
+//	resourceName := "azurerm_analysis_services_server.test"
+//	ri := tf.AccRandTimeInt()
+//	preAdminUsers := []string{"admin@domain.tld"}
+//	postAdminUsers := []string{"admin@domain.tld", "admin2@domain.tld"}
+//	preConfig := testAccAzureRMAnalysisServicesServer_adminUsers(ri, testLocation(), preAdminUsers)
+//	postConfig := testAccAzureRMAnalysisServicesServer_adminUsers(ri, testLocation(), postAdminUsers)
+//
+//	resource.ParallelTest(t, resource.TestCase{
+//		PreCheck:     func() { testAccPreCheck(t) },
+//		Providers:    testAccProviders,
+//		CheckDestroy: testCheckAzureRMAnalysisServicesServerDestroy,
+//		Steps: []resource.TestStep{
+//			{
+//				Config: preConfig,
+//				Check: resource.ComposeTestCheckFunc(
+//					testCheckAzureRMAnalysisServicesServerExists(resourceName),
+//					resource.TestCheckResourceAttr(resourceName, "admin_users.#", "1"),
+//					resource.TestCheckResourceAttr(resourceName, "admin_users.0", preAdminUsers[0]),
+//				),
+//			},
+//			{
+//				Config: postConfig,
+//				Check: resource.ComposeTestCheckFunc(
+//					testCheckAzureRMAnalysisServicesServerExists(resourceName),
+//					resource.TestCheckResourceAttr(resourceName, "admin_users.#", "2"),
+//					resource.TestCheckResourceAttr(resourceName, "admin_users.0", postAdminUsers[0]),
+//					resource.TestCheckResourceAttr(resourceName, "admin_users.1", postAdminUsers[1]),
+//				),
+//			},
+//		},
+//	})
+//}
+
+func testAccAzureRMAnalysisServicesServer_basic(rInt int, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_analysis_services_server" "test" {
+  name                = "acctestass%d"
```

Comment:
Could we ensure capitals work for the name here?
```suggestion
  name                = "acctestASS%d"
```

Reply:
renamed accordingly

Reply:
had to revert it as only lowercase is allowed